src/components/DownloadCsv/index.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Icon } from '@edx/paragon';
+import StatusAlert from '../../components/StatusAlert';
+
+class DownloadCsv extends React.Component {
+  renderDownloadCsvButton() {
+    const {
+      csvFetchMethod,
+      fetchCsvForMethod,
+      enterpriseId,
+      csvLoading,
+    } = this.props;
+
+    const downloadButtonIconClasses = csvLoading ? ['fa-spinner', 'fa-spin'] : ['fa-download'];
+
+    return (
+      <Button
+        className={['btn-outline-primary', 'download-btn']}
+        disabled={csvLoading}
+        label={
+          <span>
+            <Icon className={['fa', 'mr-2'].concat(downloadButtonIconClasses)} />
+            Download report (CSV)
+          </span>
+        }
+        onClick={() => fetchCsvForMethod(enterpriseId, csvFetchMethod)}
+      />
+    );
+  }
+
+  renderCsvErrorMessage() {
+    return (
+      <StatusAlert
+        className={['mt-3']}
+        alertType="danger"
+        iconClassName={['fa', 'fa-times-circle']}
+        title="Unable to Generate CSV Report"
+        message={`Please try again. (${this.props.csvError.message})`}
+      />
+    );
+  }
+
+  render() {
+    const {
+      csvError,
+    } = this.props;
+
+    return (
+      <div className="col">
+        <div className="row">
+          <div className="col-12 col-md-6 text-md-right">
+            {this.renderDownloadCsvButton()}
+          </div>
+        </div>
+        {csvError && this.renderCsvErrorMessage()}
```

Comment:
Similar to the comment above, the `DownloadCsv` component shouldn't render out the error message. Layout and the rendering of the CSV error message should be handled by the `Admin` component like it does now.